validator/restapi/src/main/java/tech/pegasys/teku/validator/restapi/apis/schema/DeletionStatus.java

```diff
@@ -0,0 +1,21 @@
+/*
+ * Copyright 2021 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.teku.validator.restapi.apis.schema;
+
+public enum DeletionStatus {
+  deleted,
+  not_active,
+  not_found,
+  error
```

Comment:
Not a huge fan of breaking the java naming convention here.  Not having to use funny named variables was one of the big advantages of not using the reflection approach.

I wonder if it makes sense to use `DELETED` but override `toString()` to return the external type (`deleted`).  Would need to ensure we use `toString()` instead of `name()` and the deserialiser would have to basically reimplement valueOf.  Could be a follow up PR I guess, I'm just keen to avoid the same downsides we had in the previous approach.

Reply:
I made a bit of a mess of this change, will be a follow up.